src/app.ts

```diff
@@ -1,4 +1,45 @@
+import path from 'path';
+import logger from '../logger';
 import { User, Product } from 'models';
+import DirWatcher from 'utils/dirwatcher';
+import Importer from 'utils/importer';
+import csvConverter from 'utils/csvConverter';
+import { DIRECTORY_FILES_CHANGED } from 'constants/EventNames';
+
+const PATH_TO_DATA = path.join(__dirname, '../data');
+const WATCH_DELAY_MILLISECONDS = 2000;
 
 const user = new User();
 const product = new Product();
+const watcher = new DirWatcher();
+const importer = new Importer();
+
+watcher.watch(PATH_TO_DATA, WATCH_DELAY_MILLISECONDS);
+
+watcher.on(DIRECTORY_FILES_CHANGED, changedFiles => {
+    const asyncReadOperations = [];
+    const syncReadResults = [];
+
+    changedFiles.forEach(fileName => {
+        asyncReadOperations.push(importer.import(path.join(PATH_TO_DATA, fileName)));
```

Comment:
putting data of big size in memory isn't good design. Consider streaming

Reply:
from the other side this is how task desires that to be implemented so Im ok

Reply:
I know about streaming, good point, but I was following task requirements, guess we will meet streaming a bit later.